backends/dpdk/dpdkAsmOpt.h

```diff
@@ -159,6 +159,100 @@ class RemoveUnusedMetadataFields : public Transform {
     const IR::Node* preorder(IR::DpdkAsmProgram *p) override;
 };
 
+// This pass shorten the Identifier length
+class ShortenTokenLength : public Transform {
+    ordered_map<cstring, cstring> newNameMap;
+    ordered_map<cstring, cstring> &origNameMap;
+    static size_t count;
+    // Currently Dpdk allows Identifier of 63 char long or less
+    // including dots(.) for member exp.
+    // worst case member expression will look like
+    // 1.30.30 => 63(including dot(.))
```

Comment:
This looks too conservative for metadata struct fields where the access is of the form "m.<fieldname>"
Many of the metadata fields (most of the std PSA/PNA metadata ) is being changed because of this logic. This is something not desirable.
Maybe we should handle metadata and header struct fields spearately.

Reply:
tried doing same